Test/git-issues/git-issue1495.dfy

```diff
@@ -0,0 +1,95 @@
+// RUN: %dafny /compile:0 /rprint:"%t.rprint" "%s" > "%t"
+// RUN: %diff "%s.expect" "%t"
+include "../libraries/src/Wrappers.dfy"
+import opened Wrappers
+
+datatype Bar = Bar(i: string)
+function method ParseBar(s: string): Result<Bar, string> {
+   Success(Bar(s))
+}
+
+class Foo {
+  const bar: Bar
+
+  constructor (
+    barLike: string
+  )
+    requires ParseBar(barLike).Success?
+  {
+     // Before (this) was assigned to a temporary variable.
+     (this).bar :- assert ParseBar(barLike);
+  }
+}
+
+// The fix for above should not remove the temporary variable.
+// The following test should still pass.
+class Wrapper {
+  var arg: string
+  constructor() {
+    arg := "default";
+  }
+}
+
+class Foo2 {
+  var leftWrapper: Wrapper;
+  var rightWrapper: Wrapper;
+  var currentWrapper: Wrapper;
+  var isCurrentWrapperLeft: bool;
+  ghost var Repr: set<object>;
```

Comment:
Dafny does allow an optional `;` after a field declaration, but it's cleaner and better style to omit these `;`s.